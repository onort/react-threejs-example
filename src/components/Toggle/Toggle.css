.container {
  display: flex;
  align-items: center;
  margin: 0;
  opacity: 0.7;
}

.switch {
  position: relative;
  display: inline-block;
  width: 48px;
  height: 28px;
}

.switch input {
  opacity: 0;
  width: 0;
  height: 0;
}

.slider {
  position: absolute;
  cursor: pointer;
  border-radius: 28px;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #c4c4c4;
  transition: 0.4s;
}

.slider:before {
  position: absolute;
  border-radius: 50%;
  content: "";
  height: 20px;
  width: 20px;
  left: 4px;
  bottom: 4px;
  background-color: #ffffff;
  transition: 0.4s;
}

input:checked + .slider {
  background-color: var(--primary-color);
}

input:checked + .slider:before {
  transform: translateX(20px);
}
