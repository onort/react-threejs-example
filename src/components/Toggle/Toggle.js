import React from "react"
import PropTypes from "prop-types"

import styles from "./Toggle.css"

const Toggle = props => {
  const { checked, onToggle } = props
  return (
    <div className={styles.container}>
      <label className={styles.switch}>
        <input type="checkbox" defaultChecked={checked} onChange={onToggle} />
        <span className={styles.slider} />
      </label>
    </div>
  )
}

Toggle.proptypes = {
  checked: PropTypes.bool,
  onToggle: PropTypes.func.isRequired
}

Toggle.defaultProps = {
  checked: false
}

export default Toggle
