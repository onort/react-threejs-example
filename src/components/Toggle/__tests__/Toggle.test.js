import React from "react"
import { shallow } from "enzyme"

import Toggle from "../"

describe("<Toggle />", () => {
  it("should match snaphot", () => {
    const fn = jest.fn()
    const wrapper = shallow(<Toggle onToggle={fn} checked={false} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should fire onChange event if a change occurs", () => {
    const onToggle = jest.fn()
    const changeEvent = { target: { checked: true } }
    const wrapper = shallow(<Toggle onToggle={onToggle} checked={false} />)
    wrapper.find("input").simulate("change", changeEvent)
    expect(onToggle).toBeCalledTimes(1)
    expect(onToggle).toBeCalledWith(changeEvent)
  })
})
