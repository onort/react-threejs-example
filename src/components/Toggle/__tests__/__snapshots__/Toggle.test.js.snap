// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should match snaphot 1`] = `
<div
  className="container"
>
  <label
    className="switch"
  >
    <input
      defaultChecked={false}
      onChange={[MockFunction]}
      type="checkbox"
    />
    <span
      className="slider"
    />
  </label>
</div>
`;

exports[`<Toggle /> should match snaphot 2`] = `"<div class=\\"container\\"><label class=\\"switch\\"><input type=\\"checkbox\\"/><span class=\\"slider\\"></span></label></div>"`;
